addons/l10n_id/models/res_bank.py

```diff
@@ -76,3 +101,33 @@ def _get_qr_code_generation_params(self, qr_method, amount, currency, debtor_par
                 'value': self._get_qr_vals(qr_method, amount, currency, debtor_partner, free_communication, structured_communication),
             }
         return super()._get_qr_code_generation_params(qr_method, amount, currency, debtor_partner, free_communication, structured_communication)
+
+    def _qris_fetch_status(self, qr_data):
+        """
+        using self and the given data, fetches the status of a specific QR code generated by QRIS
+        Expected values in the qr_data dict are:
+            - invoice_id returned when generating a QR code
+            - the amount present in the qr code
+            - the datetime at which the QR code was generated
+        """
+        return self._make_qris_request(QRIS_FETCH_STATUS_URL, {
+            'do': 'checkStatus',
+            'apikey': self.l10n_id_qris_api_key,
+            'mID': self.l10n_id_qris_mid,
+            'invid': qr_data['qris_invoice_id'],
+            'trxvalue': qr_data['qris_amount'],
+            'trxdate': qr_data['qris_creation_datetime'],
+        })
+
+    @api.model
+    def _make_qris_request(self, url, params):
+        """ Make an API request to QRIS, using the given url and params. """
+        try:
+            res = requests.get(url, params=params, timeout=QRIS_TIMEOUT)
```

Comment:
What if we get a `400` or `500`?
You probably need to call `res.raise_for_status()` and managed the exceptions